plugins/org.jkiss.dbeaver.model.sql.jdbc/src/org/jkiss/dbeaver/model/sql/db/InternalDB.java

```diff
@@ -122,25 +122,29 @@ protected void initSchema(@NotNull DBRProgressMonitor monitor, @NotNull Connecti
 
     protected void updateSchema(
         @NotNull DBRProgressMonitor monitor,
-        @NotNull Connection connection,
-        @NotNull ClassLoader classLoader,
-        @NotNull SQLSchemaVersionManager versionManager
+        @NotNull Connection connection
     ) throws DBException {
-        SQLSchemaManager schemaManager = new SQLSchemaManager(
-            schemaConfig.schemaId(),
-            new ClassLoaderScriptSource(
-                classLoader,
-                schemaConfig.createScriptPath(),
-                schemaConfig.updateScriptPrefix()
-            ),
-            monitor1 -> connection,
-            versionManager,
-            dialect,
-            schemaConfig.schemaVersionActual(),
-            schemaConfig.schemaVersionObsolete(),
-            databaseConfig
-        );
-        schemaManager.updateSchema(monitor);
+
+        List<SQLSchemaConfig> schemaConfigList = getSchemaConfigList();
+        for (int i = 0; i < schemaConfigList.size(); i++) {
+            SQLSchemaConfig schemaConfig = schemaConfigList.get(i);
+            SQLSchemaManager schemaManager = new SQLSchemaManager(
+                schemaConfig.schemaId(),
+                new ClassLoaderScriptSource(
+                    schemaConfig.classLoader(),
+                    schemaConfig.createScriptPath(),
+                    schemaConfig.updateScriptPrefix()
+                ),
+                monitor1 -> connection,
+                schemaConfig.versionManager(),
+                dialect,
+                schemaConfig.schemaVersionActual(),
+                schemaConfig.schemaVersionObsolete(),
+                databaseConfig,
+                this
+            );
+            schemaManager.updateSchema(monitor);
```

Comment:
Let's catch errors and rethrow with additional info (which particular schema update has failed)